modulemd/modulemd-compression.c

```diff
@@ -0,0 +1,277 @@
+  if (magic == NULL)
+    {
+      g_set_error (error,
+                   MODULEMD_ERROR,
+                   MODULEMD_ERROR_MAGIC,
+                   "magic_open() failed: Cannot allocate the magic cookie");
+      return MODULEMD_COMPRESSION_TYPE_DETECTION_FAILED;
+    }
+
+  if (magic_load (magic, NULL) == -1)
+    {
+      g_set_error (error,
+                   MODULEMD_ERROR,
+                   MODULEMD_ERROR_MAGIC,
+                   "magic_load() failed: %s",
+                   magic_error (magic));
+      return MODULEMD_COMPRESSION_TYPE_DETECTION_FAILED;
+    }
+
+  mime_type = magic_descriptor (magic, magic_fd);
+  close(magic_fd);
+
+  if (mime_type)
+    {
+      g_debug (
+        "%s: Detected mime type: %s (%s)", __func__, mime_type, filename);
+
+      if (g_str_has_prefix (mime_type, "application/x-gzip") ||
+          g_str_has_prefix (mime_type, "application/gzip") ||
+          g_str_has_prefix (mime_type, "application/gzip-compressed") ||
+          g_str_has_prefix (mime_type, "application/gzipped") ||
+          g_str_has_prefix (mime_type, "application/x-gzip-compressed") ||
+          g_str_has_prefix (mime_type, "application/x-compress") ||
+          g_str_has_prefix (mime_type, "application/x-gzip") ||
+          g_str_has_prefix (mime_type, "application/x-gunzip") ||
+          g_str_has_prefix (mime_type, "multipart/x-gzip"))
+        {
+          type = MODULEMD_COMPRESSION_TYPE_GZ_COMPRESSION;
+        }
+
+      else if (g_str_has_prefix (mime_type, "application/x-bzip2") ||
+               g_str_has_prefix (mime_type, "application/x-bz2") ||
+               g_str_has_prefix (mime_type, "application/bzip2") ||
+               g_str_has_prefix (mime_type, "application/bz2"))
+        {
+          type = MODULEMD_COMPRESSION_TYPE_BZ2_COMPRESSION;
+        }
+
+      else if (g_str_has_prefix (mime_type, "application/x-xz"))
+        {
+          type = MODULEMD_COMPRESSION_TYPE_XZ_COMPRESSION;
+        }
+
+      else if (g_str_has_prefix (mime_type, "text/plain") ||
+               g_str_has_prefix (mime_type, "text/x-yaml") ||
+               g_str_has_prefix (mime_type, "application/x-yaml"))
+        {
+          type = MODULEMD_COMPRESSION_TYPE_NO_COMPRESSION;
+        }
+    }
+  else
+    {
+      g_set_error (error,
+                   MODULEMD_ERROR,
+                   MODULEMD_ERROR_MAGIC,
+                   "mime_type() detection failed: %s",
+                   magic_error (magic));
+      return MODULEMD_COMPRESSION_TYPE_DETECTION_FAILED;
+    }
+  /* Reset the file descriptor to the start of the file, if it has moved */
+  lseek (fd, 0, SEEK_SET);
+#endif /* HAVE_LIBMAGIC */
+
+  return type;
+}
+
+ModulemdCompressionTypeEnum
+modulemd_compression_type (const gchar *name)
+{
+  if (!name)
+    return MODULEMD_COMPRESSION_TYPE_UNKNOWN_COMPRESSION;
+
+  int type = MODULEMD_COMPRESSION_TYPE_UNKNOWN_COMPRESSION;
+
+  if (!g_strcmp0 (name, "gz") || !g_strcmp0 (name, "gzip"))
```

Comment:
`modulemd_detect_compression()` also recognizes "gunzip" as a valid suffix for `MODULEMD_COMPRESSION_TYPE_GZ_COMPRESSION`. Should that suffix be included here as well for consistency?

Reply:
Agreed